### PR TITLE
feat: describe how strings with secrets are interpreted

### PR DESCRIPTION
**Pull request by keynslug**
Part of [EMQX-10808](https://emqx.atlassian.net/browse/EMQX-10808).

### CODE REVIEW

File: en_US/configuration/configuration.md [hunk 1]
```diff
@@ -274,9 +273,25 @@ The primitive types are mostly self-describing, so there is usually not a lot to
...
+
+Specialized types are essentially primitive types with additional meaning that is reflected in their names, validation rules, and descriptions.
+
+* `duration` (`duration_s` / `duration_ms`)
+
+    A string that represents a time duration, for example: `10s`, `2.5m`, `1h30m`, `2345ms`, `1W2D`. When precision is specified, finer portions of the duration may be ignored: writing `1200ms` for `duration_s` is equivalent to writing `1s`. It doesn't matter if units are in upper or lower case.
+
+* `bytesize`
+
+    A string that represents a number of bytes, for example: `10B`, `640kb`, `4MB`, `1GB`. Units are interpreted as powers of 1024, and the unit part is case-insensitive.
```
Comment (zmstone) [inline en_US/configuration/configuration.md:286]: no longer needed after https://github.com/emqx/emqx/pull/11936/files#diff-a7824deab40445e98a6a9f50b8c0e9607804f8f2383a12b52957736ac1bac3c6R146-R181

Comment (keynslug) [reply en_US/configuration/configuration.md:286]: Hmm, do these changes mean that there will no longer be `Type: duration` in the configuration manual? I'd argue that a kind of user-friendly type-level description for such types would still be warranted. Would be cool if you could click on the `duration` type name in the doc (or hover some ℹ️) and read it. Right now it seems too cumbersome and error-prone to mention what secret is and what `file://` prefix means under each field description instead, as opposed to only once for the `secret` type.

Comment (Meggielqk) [reply en_US/configuration/configuration.md:286]: Hi, have any conclusions been made about the contents in this section "Specialized Data Types"? Shall I apply the same changes to the Chinese version based on the current English content?

**Comment by zmstone** [comment]
After the new schema dump json is released, the type descriptions in the doc should have been updated (but not yet).

For Secrets, we can maybe change it here:
https://github.com/emqx/emqx/blob/f5e184ec246578e34db2a98909a5bcbf9490603e/apps/emqx_conf/src/emqx_conf_schema_types.erl#L268-L273

From `String()` to `Secret()`
And add a description about `Secret()` type in this doc.

**Pull request closed without merging**